contexts/ThreadViewContext.tsx

```diff
@@ -155,62 +150,56 @@ const getNextView = ({
   queryParams: ViewQueryParamsType;
   isNew: boolean;
   hasUpdates?: boolean;
-}): ThreadViewMode & ThreadFilters => {
-  const filters = {
-    activeFilters:
-      queryParams.filter?.filter(
-        (category): category is string => category !== null
-      ) || null,
-    excludedNotices:
-      queryParams.excludedNotices?.filter(
-        (notice): notice is string => notice !== null
-      ) || null,
-  };
-
+}): ThreadViewMode => {
   switch (nextViewMode) {
     case THREAD_VIEW_MODE.THREAD:
       return {
         threadViewMode: THREAD_VIEW_MODE.THREAD,
         galleryViewMode: null,
         timelineViewMode: null,
-        ...filters,
       };
     case THREAD_VIEW_MODE.MASONRY: {
       const currentMode = isGalleryViewQueryParams(queryParams)
         ? getQueryParamsGalleryViewMode(queryParams)?.mode
         : null;
+      log("next view, masonry currentMode", currentMode);
+
       const currentShowCover = isGalleryViewQueryParams(queryParams)
         ? getQueryParamsGalleryViewMode(queryParams)?.showCover
         : null;
+      log("next view, masonry currentShowCover", currentShowCover);
+
       return {
         threadViewMode: THREAD_VIEW_MODE.MASONRY,
         galleryViewMode: {
           mode:
+            // I'm not sure what this comment was supposed to mean/if that's correct
```

Comment:
The chained conditionals weren't working with `??`.
Because the `??` gets evaluated first, so when `currentMode` existed, it was being treated as `currentMode ? ? GALLERY_VIEW_SUB_MODE.NEW : GALLERY_VIEW_SUB_MODE.ALL` and you couldn't get to ALL mode at all.

Reply:
Oooooohhh, my bad! Good catch.